src/components/Pressable/utils.ts

```diff
@@ -0,0 +1,130 @@
+import { Insets } from 'react-native';
+import { LongPressGestureHandlerEventPayload } from '../../handlers/LongPressGestureHandler';
+import {
+  TouchData,
+  GestureStateChangeEvent,
+  GestureTouchEvent,
+} from '../../handlers/gestureHandlerCommon';
+import { HoverGestureHandlerEventPayload } from '../../handlers/gestures/hoverGesture';
+import { PressEvent, PressableEvent } from './PressableProps';
+
+const numberAsInset = (value: number): Insets => ({
+  left: value,
+  right: value,
+  top: value,
+  bottom: value,
+});
+
+const addInsets = (a: Insets, b: Insets): Insets => ({
+  left: (a.left ?? 0) + (b.left ?? 0),
+  right: (a.right ?? 0) + (b.right ?? 0),
+  top: (a.top ?? 0) + (b.top ?? 0),
+  bottom: (a.bottom ?? 0) + (b.bottom ?? 0),
+});
+
+const touchToPressEvent = (
+  data: TouchData,
+  timestamp: number,
+  targetId: number
+): PressEvent => ({
+  identifier: data.id,
+  locationX: data.x,
+  locationY: data.y,
+  pageX: data.absoluteX,
+  pageY: data.absoluteY,
+  target: targetId,
+  timestamp: timestamp,
+  touches: [], // intentionally empty
+  changedTouches: [], // intentionally empty
+});
+
+const changeToTouchData = (
+  event: GestureStateChangeEvent<
+    HoverGestureHandlerEventPayload | LongPressGestureHandlerEventPayload
+  >
+): TouchData => ({
+  id: event.handlerTag,
+  x: event.x,
+  y: event.y,
+  absoluteX: event.absoluteX,
+  absoluteY: event.absoluteY,
+});
+
+const isTouchWithinInset = (
+  dimensions: { width: number; height: number },
+  inset: Insets,
+  touch?: TouchData
+) =>
+  (touch?.x ?? 0) < (inset.right ?? 0) + dimensions.width &&
+  (touch?.y ?? 0) < (inset.bottom ?? 0) + dimensions.height &&
```

Comment:
It's an object which contains the dimensions of a `ReactNode`, and so i think the name `dimensions` is quite fitting.

It's quite a small function, with type specified 1 line above, and alternatives like `size` also partially used by react - `ScaledSize`.

So I don't think changing it will make it any clearer.

Reply:
`size` sounds more generic so I think it would be fine. But it's up to you, `dimensions` is also ok 😅